### DISCUSSION TITLE
How to insert html with inline styles in the editor?

### DISCUSSION THREAD

**Discussion by SadEaston** [ueberdosis/tiptap#4948] [General Questions]
I would like to insert the following html string into the editor, I've tried many ways and it didn't happen, does anyone have a good idea?
for example:
```
<div style="background-image:url(http://file.png);width:200px;height:200px">
  <p style="color: red; text-align: right">text</p>
</div>
```

**Comment by EskelCz** [comment]
@lujdong Have you figured it out? I'm looking for the same exact thing.

**Comment by maggie-chintaro** [comment]
I am also stuck on this. I understand it's a feature of tiptap that the styles are stripped. I believe it is possible if you update the schema. I am struggling though. If anyone has figured it out please help!

**Comment by EskelCz** [comment]
@maggie-chintaro  This is how I solved it for pasting html (it preserves html5 style attributes):
```
editorProps: {
  handlePaste: function (view, event, slice) { // handle pasting of text and images
    if (event.clipboardData.types.indexOf('text/plain') !== -1) { // parse plain text format (possibly with HTML content)
      let text = event.clipboardData.getData('text/plain')
      text = text.replace(/\n/g, '<br>') // replace newlines with line breaks
      if (editor.isEmpty) {
        editor.commands.insertContentAt(0, text, { parseOptions: { preserveWhitespace: true } })
      } else {
        editor.commands.insertContent(text, { parseOptions: { preserveWhitespace: true } })
      }
      event.preventDefault()
      return true
    }
  }
}
```
There might be a simpler way for some of the details but I don't know TipTap well enough yet. :-/
